ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java

```diff
@@ -411,6 +430,57 @@ public synchronized void reset() {
         db = null;
     }
 
+    /**
+     * Checks if the datasources table in drawable DB is stale.
+     * 
+     * @return true if datasources table is stale
+     */
+    boolean isDataSourcesTableStale() {
+        
+        // no current case open to check
+        if ((null == db) || (null == sleuthKitCase)) {
```

Comment:
These two instance variables are accessed by multiple threads, but they are only partially guarded using the intrinsic monitor of the ImageGalleryController object. The most basic solution is to be sure that we never access these variables directly unless we are holding the lock, which is easily done by ALWAYS calling the properly synchronized getDatabase and the corresponding getter for the case database reference. I am a bit concerned that we rely on finalize to close the drawable db rather than closing it when the current case is closed, so please make a story about this so we can look into the consequences later, thanks!